module3/lessons/week3/CreatingRelatedResources.md

```diff
@@ -0,0 +1,349 @@
+    All of that movie's reviews (with a rating and review content)
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```c#
+[Fact]
+public async Task Index_ReturnsViewWithReviews()
+{
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie spaceballs = new Movie { Genre = "Comedy", Title = "Spaceballs" };
+    Review review1 = new Review { Rating = 5, Content = "Better than Star Wars" };
+    Review review2 = new Review { Rating = 4, Content = "Good. But, when will then be now?" };
+    spaceballs.Reviews.Add(review1);
+    spaceballs.Reviews.Add(review2);
+    Movie youngFrankenstein = new Movie { Genre = "Comedy", Title = "Young Frankenstein" };
+    context.Movies.Add(spaceballs);
+    context.Movies.Add(youngFrankenstein);
+    context.SaveChanges();
+
+    var response = await client.GetAsync($"/Movies/{spaceballs.Id}/Reviews");
+    //Make sure the route exists!
+    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+    var html = await response.Content.ReadAsStringAsync();
+
+    //Make sure the page contains the correct info
+    Assert.Contains(spaceballs.Title, html);
+    Assert.DoesNotContain(youngFrankenstein.Title, html);
+}
+```
+
+</section>
+
+### To make these tests pass, we must:
+1. Create a Review Model
+2. Update our Movie Model
+3. Update our Context
+4. Update our Database
+    * Create a Migration
+    * Update the Database
+5. Add a controller action
+6. Add the view
+
+<section class='instructor-notes' markdown='1'>
+Walk through these steps as a group; gather steps from students as you go!
+
+We want students to understand how we can send the information we need to show information from two resources on a single view
+</section>
+
+### Multiple Resources In One Controller Action
+
+We often need to send information to a view that is related to multiple resources.  In our MvcMovie application, we need information about a movie, and information about reviews to satisfy our user story.
+
+There are a few ways to get the correct information into the views.  Two simple ways are to:
+* use information from the route to find the correct resources
+* use ViewData to send static information to the view
+
+Using these strategies will result in a controller action that looks like this:
+
+```c#
+// GET: /movies/:movieId/reviews
+[Route("Movies/{movieId:int}/reviews")]
+public IActionResult Index(int movieId)
+{
+    var movie = _context.Movies
+        .Where(m => m.Id == movieId)
+        .Include(m => m.Reviews)
+        .First(); 
+    var reviews = movie.Reviews;
+    
+    ViewData["MovieTitle"] = movie.Title;
+    return View(reviews);
+}
+```
+
+## Using a Form to Create Reviews
+
+Similar to the form we use to create movies, we are going to use a form to create reviews.  
+
+<section class='call-to-action' markdown='1'>
+
+**In Small Groups**
+Outline the components we will need in place to make the user stories below.  Be sure to consider:
```

Comment:
I love having students think about this in groups! I think it will be really productive discussions. 

This does feel like a lot of questions to keep track of. What do you think about breaking this up into two smaller discussions? 

One for:
What controller actions do we need?
What views do we need?
What data needs to be passed to and from the views?
Will we need to make any adjustments to our current implementation?

And then after you share as a group about those, then moving on to 
What tests will we add?
Will we need to make any adjustments to our current tests?